firmware/applications/SignBuddy/comms.c

```diff
@@ -0,0 +1,170 @@
+#include "comms.h"
+
+#include "ble_uart.h"
+#include "board.h"
+#include "common.h"
+#include "crc.h"
+#include "fifo.h"
+#include "logger.h"
+#include "pb_encode.h"
+#include "SignBuddy.pb.h"
+
+#include <stdlib.h>
+
+#define COMMS_TX_BUFFER_SIZE    512
+
+typedef struct __attribute__((__packed__)) {
+  uint8_t length;
+  uint8_t sample[Sample_size];
+  uint32_t crc;
+} packet_t;
+
+typedef struct {
+  TaskHandle_t      task_handle;
+  Sample            sample;
+  packet_t          packet;
+  uint8_t           sample_ready;
+  uint8_t           packet_ready;
+  uint8_t           tx_buffer[COMMS_TX_BUFFER_SIZE];
+  fifo_t            tx_fifo;
+  SemaphoreHandle_t sample_mutex;
+} state_t;
+
+static state_t s;
+
+static inline void set_sample_ready(void)
+{
+  s.sample_ready = 1;
+}
+
+static inline void clear_sample_ready(void)
+{
+  s.sample_ready = 0;
+}
+
+static inline void set_packet_ready(void)
+{
+  s.packet_ready = 1;
+}
+
+static inline void clear_packet_ready(void)
+{
+  s.packet_ready = 0;
+}
```

Comment:
what is the point of these functions?

wouldn't something like 


    s.packet_ready = TRUE;

be easier to read?

no need to change just actually wondering

Reply:
I can switch it there's no real point to the functions